flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -82,6 +82,9 @@ public class ScanContext implements Serializable {
   private static final ConfigOption<Boolean> INCLUDE_COLUMN_STATS =
       ConfigOptions.key("include-column-stats").booleanType().defaultValue(false);
 
+  private static final ConfigOption<Integer> MAX_SNAPSHOT_COUNT_PER_MONITOR_INTERVAL =
+      ConfigOptions.key("max-snapshot-count-per-monitor-interval").intType().defaultValue(Integer.MAX_VALUE);
```

Comment:
I don't think it is very clear what this is setting from the name. Not many people know Flink internals well enough to understand what the "monitor interval" is. Is there a simpler name?

What about "max-planning-group-size" or "snapshot-group-limit"?

Reply:
`max-planning-group-size` lack information about what things are in the group, `group` can be used to replace monitor interval. How about `max-snapshot-count-per-group` or `max-snapshots-per-group`?

Reply:
`max-planning-group-size` lacks information about the group items, `snapshot-group-limit` looks better to me. "group" is more concise than "per-monitor-interval".

Reply:
I agree with @chenjunjiedada that `group` is vague. what about `max-planning-snapshot-count`?

Reply:
I meant `group` is better than `per-monitor-interval`. I'm OK with both since I think we definitely need a doc to describe what it is and how it impacts the planning and the backpressure.

Reply:
I found `snapshot-group-limit` less intuitive. what does the `group` mean in this context. `monitor-interval` is an existing config that user is already familiar.

I do agree that we should make the config name clear. Just brainstorming more names. what about `max-snapshot-count-per-incremental-scan` or `max-snapshot-count-per-planning`?

Reply:
Both `max-planning-snapshot-count` and `max-snapshot-count-per-incremental-scan` are OK with me. @kbendick @rdblue what do you think?